### PR TITLE
templates

### PR DESCRIPTION
**Pull request by odinuv**
when a project template (file with tag \_python_sandbox_template_ or \_r_sandbox_template_)  is present in the project load that template and merge it with the script.
User template is also supported - the file should also have tag equal to the token description (john.doe@kompany.kom)

### CODE REVIEW

**Comment by odinuv** [comment]
merge after https://github.com/keboola/docker-jupyter/pull/3

File: src/ScriptProcessor.php [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    private function downloadFile(string $fileId): string
+    {
+        $options = new GetFileOptions();
+        $options->setFederationToken(true);
+        $fileInfo = $this->client->getFile($fileId, $options);
+
+        // Initialize S3Client with credentials from Storage API
+        $s3Client = new S3Client([
+            'version' => '2006-03-01',
```
Comment (pivnicek) [inline src/ScriptProcessor.php:105]: Do we want to have a centralised constant somewhere for this?

File: src/ScriptProcessor.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                'key' => $fileInfo['credentials']['AccessKeyId'],
+                'secret' => $fileInfo['credentials']['SecretAccessKey'],
+                'token' => $fileInfo['credentials']['SessionToken'],
+            ],
+            'http' => [
+                'decode_content' => false,
+            ],
+        ]);
+
+        $tmpFileName = sys_get_temp_dir() . '/' . uniqid('data-loader');
```
Comment (pivnicek) [inline src/ScriptProcessor.php:118]: how about using `Keboola\Temp` here?

File: src/ScriptProcessor/RTemplateAdapter.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    public function getDestinationFile(string $dataDir): string
+    {
+        // this name must match the one expected in https://github.com/keboola/docker-rstudio
+        return $dataDir . '/main.R';
+    }
+
+    public function processTemplate(string $template, string $script): string
+    {
+        $template = trim($template . "\n\n" . $script);
+        // RScript shouldn't be completely empty, because EOS is null in R
```
Comment (pivnicek) [inline src/ScriptProcessor/RTemplateAdapter.php]: how about putting there a null byte `\0`?

**Review by pivnicek — APPROVED** [review]
Wonderful!  Very understandable and thorough tests.  I just left a couple of comments about nothing very important :-)

File: src/ScriptProcessor.php (hunk 1 above)
Comment (odinuv) [reply src/ScriptProcessor.php:105]: probably not, if the api changes, you'd have to change the code too - an alternative is to use `latest` but we don't use it to make controlled updates (if they are ever required)

File: src/ScriptProcessor/RTemplateAdapter.php (hunk 3 above)
Comment (odinuv) [reply src/ScriptProcessor/RTemplateAdapter.php]: good point, i tested it again and turned out that it seems it's not needed at all, so i removed the entire hack :)

File: src/ScriptProcessor.php (hunk 2 above)
Comment (odinuv) [reply src/ScriptProcessor.php:118]: i thought about it but then it seemed like overkill - i only need a single temp file

**Pull request merged**